src/NUnitFramework/framework/Constraints/DelayedConstraint.cs

```diff
@@ -35,9 +35,129 @@ namespace NUnit.Framework.Constraints
     ///</summary>
     public class DelayedConstraint : PrefixConstraint
     {
+        /// <summary>
+        /// 
+        /// </summary>
+        public class DelayedConstraint3 : DelayedConstraint
+        {
+            private DelayedConstraint1 _parent;
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="parent"></param>
+            public DelayedConstraint3(DelayedConstraint1 parent)
+                : base(parent.BaseConstraint, parent._delayInMilliseconds, parent._pollingInterval)
+            {
+                _parent = parent; ;
+            }
+
+            /// <summary>
+            /// 
+            /// </summary>
+            public DelayedConstraint Minute
+            {
+                get
+                {
+                    return _parent;
+                }
+            }
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public class DelayedConstraint2 : DelayedConstraint
+        {
+            private readonly DelayedConstraint1 _parent;
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="parent"></param>
+            /// <exception cref="NotImplementedException"></exception>
+            public DelayedConstraint2(DelayedConstraint1 parent)
+                : base(parent.BaseConstraint, parent._delayInMilliseconds, parent._pollingInterval)
+            {
+                _parent = parent;
+                _delayInterval = parent._delayInterval;
+            }
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="milliSeconds"></param>
+            /// <returns></returns>
+            public DelayedConstraint3 PollEvery(int milliSeconds)
+            {
+                _parent._pollingInterval = milliSeconds;
+                return new DelayedConstraint3(_parent);
+            }
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public class DelayedConstraint1 : DelayedConstraint
```

Comment:
We should think up some better names. Maybe...

Existing | Better
--------|-------
DelayedConstraint.DelayedConstraint1 | DelayedConstraint.RawDelay
DelayedConstraint.DelayedConstraint2 | DelayedConstraint.DimensionedDelay
DelayedConstraint.DelayedConstraint3 | DelayedConstraint.RawPollingInterval

I'm not super happy with these names, so feel free to invent. They can be based on what has been specified or what has not yet been specified but it should probably be the same approach for each class.